### DISCUSSION TITLE
Unreliable speed and connection

### DISCUSSION THREAD

**Discussion by Kickbut101** [Gkiokan/ps4-remote-pkg-sender#59] [Q&A]
I'm using the OOP RPI on 9.0 firmware with goldenhen 2.3 I believe.

I get terribly inconsistent speeds and connections. The first 5 min will be a solid 500Mbps, then it will dip to 50Mbps for about 15-20min, then it will disconnect entirely and when I go to restart RPI it will continue again at the 50Mbps mark.

I saw issue [#52 ](https://github.com/Gkiokan/ps4-remote-pkg-sender/issues/52) and while restarting is a fine bandaid, it's not really a solution. I'm not asking that this become priority or for anyone to contribute extra time to this but I think this merits leaving the bug open instead of just closing it with the resolution of "restart your ps4 every 15 min".

**Comment by Gkiokan** [comment]
Hi mate. You are right. The previous issue was closed due the fact by "restarting" has fixed the issue for the initial reporter.   
However no one else had any reported issues with this. So I didn't make it any deeper investigations.  
 
I have no clue why it drops so hard after a couple mins, I can't think of any hardware issue here.  
I will check this issue on PKG-Zone and Area51 and see if anyone had the issue before and maybe a clue what happens.  

If this is an error with the RPI on the PS4 itself, it must be fixed on the "reciever side".  
There are couple of things planed for the March to be released. I can try to see if we can add more metrics to debug it down. 

Just curious, do you get the same behavior with the legacy RPI and the OOP RPI?

**Comment by Kickbut101** [comment]
> Just curious, do you get the same behavior with the legacy RPI and the OOP RPI?

@Gkiokan I can get back to you on that!

**Comment by Kickbut101** [comment]
@Gkiokan 

So I haven't seen it dip down to the 50Mbps more than that first time, but I did see a performance drop of some significance in the first run of the flatZ RPI version
<details>
	<summary>RPI - flatZ speeds</summary>
	```

	flatZ  - 37.87 GB file - 1st run

	      - 11:42am - 800-900Mbps
		  - 11:44am - 300Mbps
		  - 11:45am - 200Mbps
		  - 11:46am - 300Mbps
		  - 11:47am - 150Mbps
		  - 11:48am - 150Mbps
		  - 11:49am - 200Mbps
		  - 11:50am - 200Mbps
		  - 11:51am - 150Mbps
		  - 11:52am - 180Mbps
		  - 11:53am - 150Mbps
		  - 11:54am - 200Mbps
		  - 11:55am - 200Mbps
		  - 11:56am - 200Mbps
		  - 11:57am - 150Mbps
		  - 11:58am - 150Mbps
		  - 11:59am - 200Mbps
		  - 12:00pm - 200Mbps
		  - 12:01pm - 180Mbps
		  - 12:02pm - 200Mbps
		  - 12:03pm - 150Mpbs
		  
	flatZ - 37.87 GB file - 2nd run	  
	
		  - 13:44 - 706Mbps
		  - 13:45 - 679Mbps
		  - 13:45 - 830Mbps
		  - 13:46 - 827Mbps
		  - 13:46 - 757Mbps
		  - 13:47 - 851Mbps
		  - 13:47 - 713Mbps
		  - 13:48 - 826Mbps
		  - 13:48 - 851Mbps
		  - 13:49 - 681Mbps
		  - 13:49 - 852Mbps
		  - 13:50 - 748Mbps
		  
	flatZ - 96.8 GB file - 1st run
	
		  - 13:53 - 785Mbps
	      - 13:54 - 867Mbps
	      - 13:54 - 805Mbps
	      - 13:55 - 819Mbps
	      - 13:55 - 554Mbps
	      - 13:56 - 617Mbps
	      - 13:56 - 717Mbps
	      - 13:57 - 690Mbps
	      - 13:57 - 786Mbps
	      - 13:58 - 666Mbps
	      - 13:58 - 774Mbps
	      - 13:59 - 679Mbps
	      - 13:59 - 860Mbps
	      - 14:00 - 909Mbps
	      - 14:00 - 813Mbps
	      - 14:01 - 764Mbps
	      - 14:01 - 630Mbps
	      - 14:02 - 864Mbps
	      - 14:02 - 748Mbps
	      - 14:03 - 738Mbps
	      - 14:04 - 761Mbps
	      - 14:04 - 502Mbps
	      - 14:05 - 493Mbps
	      - 14:05 - 750Mbps
	      - 14:06 - 687Mbps
	      - 14:06 - 729Mbps
	      - 14:07 - 450Mbps
	      - 14:07 - 889Mbps
	      - 14:08 - 860Mbps
	      - 14:08 - 718Mbps
	      - 14:09 - 768Mbps
	      - 14:09 - 584Mbps
	      - 14:10 - 843Mbps
	      - 14:10 - 678Mbps
		  - Crashed? connection died? (Game still installed though)
	```
</details>
<details>
<summary>RPI - OOP - nazydark speeds</summary>
```

	OOP nazydark - 37.87 GB file - 1st run
	
			- 12:40 - 876Mbps
			- 12:41 - 848Mbps
			- 12:41 - 592Mbps
			- 12:42 - 870Mbps
			- 12:42 - 768Mbps
			- 12:43 - 774Mbps
			- 12:43 - 478Mbps
			- 12:44 - 494Mbps
			- 12:44 - 854Mbps
			- 12:45 - 824Mbps
			- 12:45 - 821Mbps
			- 12:46 - 850Mbps
			- 12:46 - 852Mbps
			- 12:47 - 851Mbps
	OOP nazydark - 37.87 GB file - 2nd run
	
			- 12:50 - 423Mbps
			- 12:50 - 820Mbps
			- 12:51 - 798Mbps
			- 12:51 - 737Mbps
			- 12:52 - 764Mbps
			- 12:52 - 828Mbps
			- 12:53 - 870Mbps
			- 12:53 - 804Mbps
			- 12:54 - 487Mbps
			- 12:54 - 839Mbps
			- 12:55 - 713Mbps
			- 12:55 - 694Mbps
			- 12:56 - 873Mbps
	OOP nazydark - 96.8 GB file - 1st run
	
			- 13:13 - 828Mbps
			- 13:13 - 808Mbps
			- 13:14 - 717Mbps
			- 13:14 - 832Mbps
			- 13:15 - 762Mbps
			- 13:15 - 724Mbps
			- 13:16 - 827Mbps
			- 13:16 - 811Mbps
			- 13:17 - 796Mbps
			- 13:17 - 774Mbps
			- 13:18 - 859Mbps
			- 13:18 - 774Mbps
			- 13:19 - 739Mbps
			- 13:19 - 777Mbps
			- 13:20 - 857Mbps
			- 13:20 - 796Mbps
			- 13:21 - 783Mbps
			- 13:21 - 839Mbps
			- 13:22 - 786Mbps
			- 13:22 - 822Mbps
			- 13:23 - 707Mbps
			- 13:23 - 802Mbps
			- 13:24 - 818Mbps
			- 13:24 - 713Mbps
			- 13:25 - 745Mbps
			- 13:25 - 772Mbps
			- 13:26 - 689Mbps
			- 13:26 - 750Mbps
			- 13:27 - 860Mbps
			- 13:27 - 830Mbps
			- 13:28 - 844Mbps
			- 13:28 - 784Mbps
	        - Crashed? connection died? (Game still installed though)
	```
</details>

OOP backporter -I couldn't get this one to run had error: "Invalid value for "taskid_id" parameter specified"

**Comment by Gkiokan** [comment]
@Kickbut101 Hey man, thanks for the details. I lost the trace a bit but I talked with a couple guys about this. 

It is as I thought. For the nature of the RPI it does download and install the file within the same process. So we guess it gets overloaded and stops some parts of the RPI and based on that it does work slower and in the worst part the RPI does crash and we can't get any further information for the status updates on the install process in my app which results in timeout ps4 not available blabla.  

So to fix this someone must take hands on the RPI (probably the latest one from nazydark) and develop it further.  
backporter didn't work on his OOP version, so as far I know nazydark has the latest updates.  
  
I am sorry for now I can't do much at this state as I have some other prios first. But I will keep that in my mind and once I have time I can get my hands on a own RPI to make it more stable.   

There are still other options to Install PKG's with the FTP  which seems to be a bit more stable. So one of the next updates will include the integrated FTP client into the PKG Sender v2 and will allow you to browse and do some fancy stuff like Install all cheat json files per click or put the PKG files into the data folder to install them via IPI.

**Comment by Gkiokan** [comment]
Actually Sam Daniels did a guide on a old version of the app how to get the best speeds (most errors in the vid are already fixed.)  Hope that helps at least a bit.  
[PKG Sender V2 Guide by Sam Daniels (v2.4.3) on YouTube](https://www.youtube.com/watch?v=RyKUm5rndPQ)

**Comment by Gkiokan** [comment]
I converted this into a Discussion. So anyone can reply to this having the same issue to help and find a way to fix it without refactoring the hole RPI (on the p4).  Probably someone of the 42k downloaders could make a Guide how to fix that special case.
